### PR TITLE
Search on timeout

### PR DESCRIPTION
**Pull request by mnunberg**

### CODE REVIEW

**Comment by mnunberg** [comment]
#479  #294

File: src/config.h [hunk 1]
```diff
@@ -122,7 +119,7 @@ sds RSConfig_GetInfoString(const RSConfig *config);
 #define RS_DEFAULT_CONFIG                                                                       \
   {                                                                                             \
     .concurrentMode = 1, .extLoad = NULL, .enableGC = 1, .minTermPrefix = 2,                    \
-    .maxPrefixExpansions = 200, .queryTimeoutMS = 500, .timeoutPolicy = TimeoutPolicy_Return,   \
+    .maxPrefixExpansions = 200, .queryTimeoutMS = 500, .timeoutPolicy = TimeoutPolicy_Fail,     \
```
Comment (MeirShpilraien) [inline src/config.h]: are we sure we want to change it? people might start getting errors that they did not get before ...

Comment (MeirShpilraien) [reply src/config.h]: @gkorland what do you think? should we change it?

**Comment by gkorland** [comment]
Let's not change the default behavior on timeout on this 1.4.1 let's push the breaking change to 2.0.0.

**Comment by mnunberg** [comment]
Ok, I’ll revert it back. This PR (without the change in default) would still allow a user to issue a query and determine if something is a timeout issue without restarting Redis, so that’s a plus, at least

**Pull request merged**